lib/gem_ext/rails/join_association.rb

```diff
@@ -0,0 +1,20 @@
+require 'active_record/associations/join_dependency/join_association'
+
+ActiveRecord::Associations::JoinDependency::JoinAssociation.class_eval do
+  def build_constraint(klass, table, key, foreign_table, foreign_key)
```

Comment:
Yes it has to have special behaviour to do an `IN` for the join instead of an `=`